Update viewer.html for Internet Explorer Compatibility

WilliamRClark opened the pull request:
If your Internet Explorer 11 default compatibility settings are set to "I.E 7 Compatibility", the PDF plugin will not load.  This fix is the same one used by AngularJS to force the browser to use HTML5 mode.

WilliamRClark committed:
Update viewer.html

If your Internet Explorer 11 default compatibility settings are set to "I.E 7 Compatibility", the PDF plugin will not load.  This fix is the same one used by AngularJS to force the browser to use HTML5 mode.

timvandermeij commented:
@WilliamRClark After addressing the comment above, please squash the resulting commits into one using https://github.com/mozilla/pdf.js/wiki/Squashing-Commits.

yurydelendik commented:
> Change this to <!-- Ignore legacy IE compatibility settings and prefer HTML5 --> to fix a typo.
@timvandermeij I think we can remove this comment.

But I think we enclose the meta inside `<!--#if GENERIC -->` to avoid polluting for Firefox and Chrome.

WilliamRClark committed:
Update source to reflect code review comments.

Comment changed to:
<!-- Ignore legacy IE compatibility settings and prefer HTML5 -->
from
<!-- Ignore legacy I.E. compatibility settings and prefer HTML5 -->
per recommendation.

WilliamRClark commented:
Comment changes committed.
 


     On Monday, August 31, 2015 7:39 PM, Yury Delendik <notifications@github.com> wrote:
   

 
Change this to <!-- Ignore legacy IE compatibility settings and prefer HTML5 --> to fix a typo.
@timvandermeij I think we can remove this comment.
But I think we enclose the meta inside <!--#if GENERIC --> to avoid polluting for Firefox and Chrome.—
Reply to this email directly or view it on GitHub.

Snuffleupagus commented on web/viewer.html:
```diff
@@ -27,6 +27,8 @@
     <meta name="google" content="notranslate">
 <!--#endif-->
+    <!-- Ignore legacy IE compatibility settings and prefer HTML5 -->
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
```
Please also address the review comments in https://github.com/mozilla/pdf.js/pull/6374#issuecomment-136528408, i.e. change the patch to:
```html
<!--#if GENERIC-->
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
<!--#endif-->
```
After this, please squash the commits, see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits.

WilliamRClark replied:
Still have to validate with Edge, the other versions of Internet Explorer.

Will commit this change when testing is complete.

WilliamRClark replied:
I still need to test this change on Edge, and I.E. 9.0 and 10.0.  Will commit once this is done. 


     On Tuesday, September 1, 2015 8:10 AM, Jonas Jenwald <notifications@github.com> wrote:
   

 In web/viewer.html:> @@ -27,6 +27,8 @@
>  <!--#if GENERIC || CHROME-->
>      <meta name="google" content="notranslate">
>  <!--#endif-->
> +    <!-- Ignore legacy IE compatibility settings and prefer HTML5 -->
> +    <meta http-equiv="X-UA-Compatible" content="IE=edge">
Please also address the review comments in #6374 (comment), i.e. change the patch to:<!--#if GENERIC-->
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
<!--#endif-->After this, please squash the commits, see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits.—
Reply to this email directly or view it on GitHub.

Rob--W commented on web/viewer.html:
```diff
@@ -27,6 +27,8 @@
 <!--#if GENERIC || CHROME-->
     <meta name="google" content="notranslate">
 <!--#endif-->
+    <!-- Ignore legacy IE compatibility settings and prefer HTML5 -->
```
I suggest to remove this comment. Everyone who is familiar with IE knows what the meta tag does. And otherwise it is just one internet search away.

WilliamRClark commented:
For maximum de-cluttering:

<!--#if IE-->
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<![endif]>

Rob--W commented:
@WilliamRClark We don't have IE-specifc builds.

WilliamRClark committed:
Address code review comments

Address code review comments
(1) Remove InternetExplorer comment
(2) Add generic build macro around http-equiv for Internet Explorer

Snuffleupagus commented on web/viewer.html:
```diff
@@ -27,6 +27,9 @@
 <!--#if GENERIC || CHROME-->
     <meta name="google" content="notranslate">
 <!--#endif-->
+<!--#if GENERIC || CHROME-->    
```
This line should read `<!--#if GENERIC-->`, since this code is *not* needed in Chrome. Also, please remove the trailing whitespace at the end of this line.

Finally, you also need to squash the commits, please see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits.

WilliamRClark replied:
Getting rid of the Chrome reference.

WilliamRClark committed:
More code review comments

Get rid of trailing space, and unneeded reference to CHROME.

WilliamRClark commented:
All code review changes should be complete.

timvandermeij commented:
@WilliamRClark Looks good. All that needs to be done is to squash the commits into one commit. See https://github.com/mozilla/pdf.js/wiki/Squashing-Commits on how to do that. We can land this patch after that has been done.

yurydelendik commented:
Merged by b1515fdd1060e3a68776b9fe022cd830f66e5f7d.

Thank you for the patch.
